src/libraries/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509Certificate2.cs

```diff
@@ -628,6 +632,347 @@ public bool Verify()
             }
         }
 
+        /// <summary>
+        /// Creates a new X509 certificate from the file contents of an RFC 7468 PEM-encoded
+        /// certificate and private key.
+        /// </summary>
+        /// <param name="certPemFilePath">The path for the PEM-encoded X509 certificate.</param>
+        /// <param name="keyPemFilePath">
+        /// If specified, the path for the PEM-encoded private key.
+        /// If unspecified, the <paramref name="certPemFilePath" /> file will be used to load
+        /// the private key.
+        /// </param>
+        /// <returns>A new certificate with the private key.</returns>
+        /// <exception cref="ArgumentException">
+        /// <para>
+        ///   The contents of the file path in <paramref name="certPemFilePath" /> do not contain
+        ///   a PEM-encoded certificate, or it is malformed.
```

Comment:
Hm.  Is there prior art for using ArgumentException for "the file didn't have what I expected"?  It feels like that'd be more of a runtime exception, so (given this is crypto) CryptographicException.

Reply:
Given that "no appropriate PEM label was found" and "the base64 contents don't actually make a valid certificate" (what'll come out of the "hey, native component, load this.  Oh, it failed? throw." path) are both "the file contents aren't valid", I think they should both be CryptographicException.